backend/common_constructs/user_pool.py

```diff
@@ -0,0 +1,149 @@
+from typing import List
+
+from aws_cdk import CfnOutput, Duration, RemovalPolicy
+from aws_cdk.aws_cognito import UserPool as CdkUserPool, UserPoolEmail, AccountRecovery, AutoVerifiedAttrs, \
+    AdvancedSecurityMode, DeviceTracking, Mfa, MfaSecondFactor, PasswordPolicy, StandardAttributes, \
+    StandardAttribute, CognitoDomainOptions, AuthFlow, OAuthSettings, OAuthFlows, ClientAttributes, \
+    CfnUserPoolRiskConfigurationAttachment, OAuthScope
+from aws_cdk.aws_kms import IKey
+from cdk_nag import NagSuppressions
+from constructs import Construct
+
+
+class UserPool(CdkUserPool):
+    def __init__(
+            self, scope: Construct, construct_id: str, *,
+            cognito_domain_prefix: str,
+            environment_name: str,
+            encryption_key: IKey,
+            removal_policy,
+            **kwargs
+    ):
+        super().__init__(
+            scope, construct_id,
+            removal_policy=removal_policy,
+            deletion_protection=removal_policy != RemovalPolicy.DESTROY,
+            email=UserPoolEmail.with_cognito(),
+            # user_invitation=UserInvitationConfig(...),
```

Comment:
This is the start of what a user pool may look like for board admins. Out of the box, Cognito user pools can send invitation, password reset, and security notification emails to users, however the default sender email and email content is very generic with no references at all to the system the user was invited to. One customization element we will want to include is to customize the email content as well as the sender so that the communications are not confusing. This comment is a place-holder to remind me exactly where that customization gets wired in, when we get to that implementation.